### PR TITLE
WRQ-22937: Updated storybook to version 8

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Storybook needed to be updated to the latest version

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated list of storybook dependencies and their version to v8.x based on the migration guide https://github.com/storybookjs/storybook/blob/next/MIGRATION.md
Changed configuration based on the storybook v7 and v8 requirements 

- main.js: `core.builder` information replaced by `framework.name`. 
- Addon '@enact/storybook-utils/addons/docs' replaced by '@storybook/addon-docs' because storybook addon for docs does not support customization anymore https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#dropped-addon-docs-manual-configuration
- story filenames different than xxx.stories.js need an indexer https://github.com/storybookjs/storybook/pull/20700/files#diff-ea5ebe92abd71b6467c58252b92c04b8d49cc140e8a94f9cf5bb7a76bf2256a4 `experimental_indexers` information was added in `main.js`
- `preview.js`: docs.inlineStories has been renamed docs.story.inline; docs.iframeHeight has been renamed  docs.story.iframeHeight
- `manager.js` changed configuration based on https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#ui-layout-state-has-changed-shape 
- `package.json` storybook dev and build commands adapted to v8 naming.
- new root container id used by storybook is called "storybook-root". This change breaks enact's logic to apply spotlight related styling in SpotlightRootDecorator https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#change-of-root-html-ids ThemeDecorator already has a rootId prop and can be set from the sampler app to `storybook-root`, but it was not passed to SpotlightRootDecorator. A PR in enact repo is needed also and is attached to the user story
- `manager.js`: The state and API for the UI layout has changed https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#ui-layout-state-has-changed-shape 

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Note: although docs is disabled, I left the docs configuration intact in preview.js. It will be needed when we activate docs and does not affect current configuration

### Links
[//]: # (Related issues, references)
WRQ-22937

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian ([daniel.stoian@lgepartner.com](mailto:daniel.stoian@lgepartner.com))

### CODE REVIEW

**Comment by adrian-cocoara-lgp** [comment]
Overall looks good to me.
There is one issue on window resize. A native HTML scrollbar is applied to globals controls.  This seems to be a Storybook bug.
On develop branch, a `os-scrollbar` is applied to globals list
image

on feature/storybook8 branch, this class does not appear
image

A discussion is needed on whether to override this class or to open a Storybook issue.

**Review by adrian-cocoara-lgp — APPROVED** [review]

**Comment by MikyungKim** [comment]
Could you please rebase your branch with the latest `feature/enact_v5`?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ The following is a curated list of changes in the Enact sandstone module, newest
 
 - `sandstone/ContextualPopupDecorator` and `sandstone/Dropdown` to have sibling DOM node as alternative to findDOMNode API which will be removed in React 19
 - `sandstone/PageViews` to hide dot page indicator when there is only one page
+- `sandstone/ThemeDecorator` to pass `rootId` to `spotlight/SpotlightRootDecorator`
```
Comment (MikyungKim) [inline CHANGELOG.md:11]: I'm not sure we should add this changelog or not.. 🤔 I'll get back to you after discussing this with members.

File: samples/sampler/.qa-storybook/main.js [hunk 2]
```diff
@@ -1,23 +1,40 @@
 /* global __dirname */
 
 const webpack = require('@enact/storybook-utils/configs/webpack');
+const { loadCsf } = require('@storybook/csf-tools');
+const { readFileSync }  = require('fs');
```
Comment (MikyungKim) [inline samples/sampler/.qa-storybook/main.js:5]:
```suggestion
const {loadCsf} = require('@storybook/csf-tools');
const {readFileSync}  = require('fs');
```

File: CHANGELOG.md (hunk 1 above)
Comment (MikyungKim) [reply CHANGELOG.md:11]: We think it's okay not to have this change log. Because it's merely using it.

File: samples/sampler/package.json [hunk 3]
```diff
@@ -24,18 +24,20 @@
     "@enact/i18n": "^4.9.0-alpha.3",
     "@enact/sandstone": "../../",
     "@enact/spotlight": "^4.9.0-alpha.3",
-    "@enact/storybook-utils": "^5.1.4",
+    "@enact/storybook-utils": "enactjs/storybook-utils#feature/storybook8",
```
Comment (MikyungKim) [inline samples/sampler/package.json:27]: After Merging https://github.com/enactjs/storybook-utils/pull/71, would you please update the branch to `feature/enact_v5`?
Also, please update the version of storybook modules to 8.1.6.

**Comment by MikyungKim** [comment]
@daniel-stoian-lgp , Have you tested storybook version 8.1.6 with this branch?
I manually updated the storybook-related modules to 8.1.6 and symlinked @enact/storybook-utils to `feature/storybook8`.
I got an error like below. 🤔 Do you get this error too?
```
Error: Cannot find module '@storybook/csf-tools'
Require stack:
- /.storybook/main.js
- /node_modules/@storybook/core-server/node_modules/@storybook/core-common/dist/index.js
- /node_modules/@storybook/core-server/dist/index.js
- /node_modules/@storybook/cli/dist/generate.js
- /node_modules/@storybook/cli/bin/index.js
- /node_modules/storybook/index.js
```

**Comment by daniel-stoian-lgp** [comment]
> @storybook/csf-tools

Hello Mikyung. I do not face these issue. I removed node_modules and npm-shinkwrap. 
Them npm i && npm shrinkwrap && enact link

image

**Comment by MikyungKim** [comment]
Hm.. I guess symlinking made some problem. If I don't symlink the error is not reproducing. Thanks.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**